### PR TITLE
Try to fix ioob for selected entries

### PR DESCRIPTION
**Pull request by Siedlerchr**
Refs #10373 

Use changed list from the listener

### Mandatory checks
- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/gui/LibraryTab.java [hunk 1]
```diff
@@ -208,7 +209,16 @@ public void cancelLoading() {
...
+     * layout
+     * to
+     * display
+     * in
+     * the
+     * tab
+     * when
+     * it's
+     * loading
      */
```
Comment (ThiloteE) [inline src/main/java/org/jabref/gui/LibraryTab.java:222]: Are you sure, comments are supposed to look like this? IMHO, it is hard to read.

Comment (Siedlerchr) [reply src/main/java/org/jabref/gui/LibraryTab.java:222]: Nope, thanks, that's the wrong formatting of intellilj

**Comment by ThiloteE** [comment]
I tried this PR. In its current state does **NOT** fix #8719

**Comment by ThiloteE** [comment]
java.lang.IndexOutOfBoundsException: [ fromIndex: 0, toIndex: 2, size: 1 ]
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.scene.control.ReadOnlyUnbackedObservableList.subList(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ListChangeListener$Change.getAddedSubList(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.scene.control.SelectedItemsReadOnlyObservableList.getAddedElements(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.scene.control.SelectedItemsReadOnlyObservableList.lambda$new$0(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.collections.ListListenerHelper$Generic.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ObservableListBase.fireChange(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ListChangeBuilder.commit(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ListChangeBuilder.endChange(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ObservableListBase.endChange(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.scene.control.ReadOnlyUnbackedObservableList._endChange(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.control.MultipleSelectionModelBase$SelectedIndicesList._endChange(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.control.ControlUtils.updateSelectedIndices(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.control.TableView$TableViewArrayListSelectionModel.fireCustomSelectedCellsListChangeEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.control.TableView$TableViewArrayListSelectionModel.lambda$new$1(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.collections.ListListenerHelper$SingleChange.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ObservableListBase.fireChange(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ListChangeBuilder.commit(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ListChangeBuilder.endChange(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ObservableListBase.endChange(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.transformation.SortedList.sourceChanged(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.transformation.TransformationList.lambda$getListener$0(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.WeakListChangeListener.onChanged(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.collections.ListListenerHelper$SingleChange.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ObservableListBase.fireChange(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ListChangeBuilder.commit(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ListChangeBuilder.endChange(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ObservableListBase.endChange(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ModifiableObservableListBase.setAll(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.scene.control.SelectedCellsMap.setAll(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.control.TableView$TableViewArrayListSelectionModel.updateSelection(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.control.TableView$TableViewArrayListSelectionModel.lambda$new$3(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.WeakListChangeListener.onChanged(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.collections.ListListenerHelper$Generic.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ObservableListBase.fireChange(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ListChangeBuilder.commit(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ListChangeBuilder.endChange(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ObservableListBase.endChange(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.transformation.SortedList.sourceChanged(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.transformation.TransformationList.lambda$getListener$0(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.WeakListChangeListener.onChanged(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.collections.ListListenerHelper$Generic.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ObservableListBase.fireChange(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ListChangeBuilder.commit(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ListChangeBuilder.endChange(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ObservableListBase.endChange(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.transformation.FilteredList.sourceChanged(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.transformation.TransformationList.lambda$getListener$0(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.WeakListChangeListener.onChanged(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.collections.ListListenerHelper$SingleChange.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ObservableListBase.fireChange(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ListChangeBuilder.commit(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ListChangeBuilder.endChange(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ObservableListBase.endChange(Unknown Source)
	at com.tobiasdiez.easybind@2.2.1-SNAPSHOT/com.tobiasdiez.easybind.MappedBackedList.sourceChanged(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.transformation.TransformationList.lambda$getListener$0(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.WeakListChangeListener.onChanged(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.collections.ListListenerHelper$SingleChange.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ObservableListBase.fireChange(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.gui.util.UiThreadList.sourceChanged(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.transformation.TransformationList.lambda$getListener$0(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.WeakListChangeListener.onChanged(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.collections.ListListenerHelper$SingleChange.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ObservableListBase.fireChange(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.FXCollections$UnmodifiableObservableListImpl.lambda$new$0(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.WeakListChangeListener.onChanged(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.collections.ListListenerHelper$Generic.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.FXCollections$SynchronizedObservableList.lambda$new$0(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.WeakListChangeListener.onChanged(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.collections.ListListenerHelper$SingleChange.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ObservableListBase.fireChange(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ListChangeBuilder.commit(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ListChangeBuilder.endChange(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ObservableListBase.endChange(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.collections.ObservableListWrapper.access$200(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.collections.ObservableListWrapper$1$1.invalidated(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.collections.MapListenerHelper$Generic.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.collections.MapListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.collections.ObservableMapWrapper.callObservers(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.collections.ObservableMapWrapper.put(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.model.entry.BibEntry.setField(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.model.entry.BibEntry.setField(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.gui.entryeditor.SourceTab.storeSource(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.gui.entryeditor.SourceTab.bindToEntry(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.gui.entryeditor.EntryEditorTab.notifyAboutFocus(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.gui.entryeditor.EntryEditor.setEntry(Unknown Source)
	at java.base/java.util.Optional.ifPresent(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.gui.LibraryTab.lambda$createMainTable$10(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.collections.ListListenerHelper$Generic.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ObservableListBase.fireChange(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ListChangeBuilder.commit(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ListChangeBuilder.endChange(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ObservableListBase.endChange(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.scene.control.SelectedItemsReadOnlyObservableList.lambda$new$0(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.collections.ListListenerHelper$Generic.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ObservableListBase.fireChange(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ListChangeBuilder.commit(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ListChangeBuilder.endChange(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.collections.ObservableListBase.endChange(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.scene.control.ReadOnlyUnbackedObservableList._endChange(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.control.MultipleSelectionModelBase$SelectedIndicesList._endChange(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.control.ControlUtils.updateSelectedIndices(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.control.TableView$TableViewArrayListSelectionModel.fireCustomSelectedCellsListChangeEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.control.TableView$TableViewArrayListSelectionModel.clearAndSelect(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.control.TableView$TableViewSelectionModel.clearAndSelect(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.scene.control.behavior.TableCellBehaviorBase.simpleSelect(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.scene.control.behavior.TableCellBehaviorBase.doSelect(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.scene.control.behavior.CellBehaviorBase.mousePressed(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.scene.control.inputmap.InputMap.handle(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventUtil.fireEventImpl(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventUtil.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.event.Event.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.Scene$MouseHandler.process(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.Scene.processMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.Scene$ScenePeerListener.mouseEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$2(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.glass.ui.View.handleMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.glass.ui.View.notifyMouse(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
	at org.jabref.merged.module@5.10.60000/com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)

**Comment by Siedlerchr** [comment]
note, that thilo's exception is more related to #10379 
with thilo's libray and instructions I could reproduce it e.g. copy field author bibtex in the entry editor to another entry while having a sort order set

[librarytoreproduce.bib.txt](https://github.com/JabRef/jabref/files/12641211/librarytoreproduce.bib.txt)

File: src/main/java/org/jabref/gui/LibraryTab.java [hunk 2]
```diff
@@ -517,15 +518,14 @@ private void createMainTable() {
                 entryTypesManager,
                 taskExecutor,
                 fileUpdateMonitor);
-
+        // conntent binding does not work as it won't trigger the needsEntries checker
```
Comment (koppor) [inline src/main/java/org/jabref/gui/LibraryTab.java]: I did not quite get the reference to `needsEntries checker`. Thus, I tried to find it:

```suggestion
        // content binding does not work as it does not trigger the ManageKeywordsAction#needsEntriesSelected checker
```

File: src/main/java/org/jabref/gui/LibraryTab.java [hunk 3]
```diff
@@ -517,15 +518,14 @@ private void createMainTable() {
...
+        mainTable.addSelectionListener(event -> stateManager.setSelectedEntries(event.getList().stream().map(BibEntryTableViewModel::getEntry).toList()));
         // Add the listener that binds selection to state manager (TODO: should be replaced by proper JavaFX binding as soon as table is implemented in JavaFX)
```
Comment (koppor) [inline src/main/java/org/jabref/gui/LibraryTab.java:521]: This comment is somehow lost now? Does it apply to the line above?

**Review by koppor — COMMENTED** [review]
While reading, I could not quite follow. Added some comments - but also pushed some changes.

File: src/main/java/org/jabref/gui/LibraryTab.java [hunk 4]
```diff
@@ -562,7 +562,7 @@ public void setupMainPanel() {
 
     /**
      * Set up auto completion for this database
-     */
+     */r
```
Comment (koppor) [inline src/main/java/org/jabref/gui/LibraryTab.java:565]:
```suggestion
     */
```

File: src/main/java/org/jabref/gui/LibraryTab.java (hunk 2 above)
Comment (Siedlerchr) [reply src/main/java/org/jabref/gui/LibraryTab.java]: I added a clarification, ActionHelper#needsEntriesSelected - enables or disables the menu items

**Review by koppor — APPROVED** [review]

**Comment by Siedlerchr** [comment]
Note that thi**s DOES NOT FIX** Thilo's reported exception.  
That happens when you modify the bibtex while having a sort order

**Comment by koppor** [comment]
+1 for follow-up. Code looks better than before, therefore merging.

**Pull request merged**